collect_app/src/main/java/org/odk/collect/android/widgets/ExStringWidget.java

```diff
@@ -265,6 +265,7 @@ public void onButtonClick(int buttonId) {
             }
         } else {
             onException(errorString);
+            setFocus(getContext());
```

Comment:
This change looks strange to me. I would rather expect something like https://stackoverflow.com/a/5106399 Could you explain that `setFocus`?
and even if it' ok it should be in `onException()` method I think.

Reply:
@grzesiek2010 if you look the `setFocus()` method, it does the same thing as given in the link. Whether or not to put `setFocus()` inside `onException()` is a doubt. The reason is that because the `onException()` is a general method of handling exceptions, which may or may not involve handling the keyboard visibility.
If you say, I can perform the changes in the PR.

Reply:
Ahhh `setFocus()` is our method. I got it. But that `setFocus()` method does more than we need. We just need to show the keyboard. So I would recommend creating a new method just for showing the keyboard then you can use it here and probably you can refactor that `setFocus()` method and use your method there too.

Reply:
@grzesiek2010 you didn't like the use of `setFocus` because of the extra check on `hasExApp` and on read-only? Those don't seem like a big deal to me and using `setFocus` is clearer.

Reply:
@lognaturel so should I use `setFocus()` or my new method?